manager/cmd/version.go

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2019 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	container "cloud.google.com/go/container/apiv1"
+	"github.com/GoogleCloudPlatform/gke-rolling-updates-demo/manager/pkg/cluster"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	master  bool
+	version string
+)
+
+// versionCmd represents the version command
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Returns proper master and node versions for the given inputs",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		client, err := container.NewClusterManagerClient(ctx)
+		if err != nil {
+			log.Fatalf("unable to create cluster manager client: %s", err)
+		}
+
+		if project == "" {
+			log.Fatalf("Must specify a project")
+		}
+
+		if location == "" {
+			log.Fatalf("Must specify a location")
+		}
+
+		if clusterName == "" {
+			log.Fatalf("Must specify a cluster name")
+		}
+
+		cluster := cluster.NewManagedCluster(client, project, location, clusterName, 0)
+
+		fmt.Printf("%v", cluster)
+
+		if master {
+			v, err := cluster.LatestMasterVersionForReleaseSeries(ctx, version)
+			if err != nil {
+				log.Fatalf("failed to get latest master versions: %s", err)
+			}
+			fmt.Fprintf(os.Stdout, v)
+		}
+
+		if cluster.Cluster == nil {
+			fmt.Fprintf(os.Stderr, "cluster doesn't exist")
+		}
+
+		fmt.Fprintf(os.Stdout, cluster.Cluster.GetCurrentMasterVersion())
```

Comment:
let's output something nice.  Like all the data we used to get this.  Also does Fprintf throw an error?

Reply:
We should talk about how we want to format the output data. I'd like to make the output as machine parseable as possible. I'd kinda like to make it friendly to Unix piping, although I'm not quite sure how to accomplish this.

As for the second question, I have a feeling you already know the answer. :P I was unaware of the error returned, although it makes sense it would throw an error now that I think about it. Fixed.